### ISSUE TITLE
[bug] Cannot resolve conflicts in conanfile.txt

### ISSUE THREAD

**Issue by madebr** [conan-io/conan#17056]
### Describe the bug

I just created the `conanfile.txt` at the bottom for a project.
When installing it. conan informs me there is a version conflict between stb/cci.20230920 and stb/cci.20240531.
Looking through the documentation, I don't see a way to resolve this in the txt file.

I believe overriding packages should be do-able from the txt files, and does not require a conversion to py files.

Also, the error messages contains: `Run conan graph info with your recipe and add --format=html to inspect the graph errors in an easier to visualize way.`
But when I add `--format=html`, no html files are generated.

### How to reproduce it

**conanfile.txt**
```
[requires]
miniz/3.0.2
nlohmann_json/3.11.3
stb/cci.20240531
libsquish/1.15
glew/2.2.0
assimp/5.4.2
fmt/11.0.2

[generators]
CMakeDeps
CMakeToolchain
```

```
$ conan install . --output-folder=build --build=missing
```

**Comment by AbrilRBS** [comment]
> I believe overriding packages should be do-able from the txt files, and does not require a conversion to py files.

The preferred way to change dependencies versions would be to use the `[replace_requires]` feature of profiles, see https://docs.conan.io/2/reference/config_files/profiles.html#replace-requires, which allow you to specify replacements for your versions, so for example:
```
[replace_requires]
dep/*: dep/1.1               # To override all dep versions in recipes to a specific version dep/1.1
```

> Also, the error messages contains: Run conan graph info with your recipe and add --format=html to inspect the graph errors in an easier to visualize way.
But when I add --format=html, no html files are generated.

Conan 2 does not generate output files by itself. You'll need to redirect the output manually to a file of your choosing, like:
`conan graph info .... --format=html > graph.html`. Said file will then have your HTML ready for viewing in a browser

Let me know if this helps, thanks!

**Comment by memsharded** [comment]
> You might want to use another version in another prefix.

Not clear what you mean, could you please clarify what is ``another prefix``?

**Comment by madebr** [comment]
The profile is a global configuration.
By prefix, I mean the prefixes of the dependencies of a project (`CMAKE_PREFIX_PATH`).

The overrides of the dependencies of projectA should not  influence the  dependencies of projectB.

**Comment by memsharded** [comment]
> The overrides of the dependencies of projectA should not influence the dependencies of projectB.

You could have 2 different profiles ``conanfileA_overrides`` and ``conanfileB_overrides`` and you use them when installing deps for the different projects:
```
conan install conanfileA.txt -pr=myprofile --pr=conanfileA_overrides 
# and
conan install conanfileB.txt -pr=myprofile --pr=conanfileB_overrides
```
Profiles doesn't need to be global and used for all projects

**Comment by madebr** [comment]
That looks useful, though it feels weird to mix dependency issues with profile.

`conan install -h` does not show a `--pr=` option. Is `--pr=` allowed?

**Comment by memsharded** [comment]
> That looks useful, though it feels weird to mix dependency issues with profile.

this is because an override is considered an exceptional case, kind of graph external configuration, not part of the ``conanfile``. The ``conanfile`` is focused on the definition of its dependencies, if there are conflicts upstream, the best approach of course would be to fix them upstream, updating the dependencies recipes. If that is not possible and a quick fix is desired, then adding it in a local profile can solve it, but keeping it in a separate profile makes it more evident and explicit that this is external and temporary configuration to resolve the conflict, not a permanent definition of the dependency graphs. 

This keeps the general design that a ``conanfile`` defines "what" dependencies it wants, while profiles and configuration define "how" (defining options, changing package_id_modes, etc).

Also, it is important to consider that overrides in ``conanfiles`` is possibly the most challenging thing in the graph resolution, build-orders, CI-flows, etc, because they are graph-local information that is defined only in  downstream nodes and as such not always available to resolve when processing some upstream nodes, this is why putting them in profiles makes things way more robust.

> conan install -h does not show a --pr= option. Is --pr= allowed?

No, sorry, my mistake, it is just ``-pr`` always

**Comment by madebr** [comment]
Thanks for the background.

I felt it was a missing feature in `conanfile.txt`, because it is possible to do in `conanfile.py` through `self.requires(... , force=True)`.
The docs only explain how to fix it in py files, not through txt files.

**Comment by memsharded** [comment]
Yes, that is true, but in ``conanfile.py`` is possible also because of historical reasons, the ``[replace_requires]`` is very modern in Conan2.X, while the ``requires(dep, force=True / override=True)`` has been there for a very long time, so not really possible to remove it. But as the recommendation is to avoid overrides as much as possible, it was not worth to invent a new custom text-based syntax in ``conanfile.txt`` for something that we know it is not great.

**Comment by madebr** [comment]
Feel free to close this issue.
I'd suggest to document this somehow to avoid explaining this to every person.

**Comment by memsharded** [comment]
Moving to the docs repo then, thanks for your feedback!
